x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java

```diff
@@ -240,4 +243,8 @@ public static DataType fromTypeName(String esType) {
             return DataType.UNSUPPORTED;
         }
     }
+
+    public String format() {
```

Comment:
Not okay conceptually and implementation wise since the base class is aware of its child type.
Make format() return DocValueFieldsContext by default and override DateTime and Date to return DATE_PARSE_FORMA - basic overriding.

Reply:
But this is an enum, I don't get it.

Reply:
Let me rephrase this - a dataType can have different formats. Currently we store that not in the `DataType` but rather inside the `DateEsField` as it can vary from field to field.
Ironically due to our usage of field_caps this information is not accessible anymore however it might be in the future.
DateEsField allows a custom format to be specified though I'm not sure this is something we need to use necessarily.

If the format could be used on `DateEsField` I would opt for that, if not, it can stay as is in the `DataType` but the two classes need to be aligned so the format is defined and stored only in one place.

Reply:
The `getFormats()` on `DateEsField` is currently not used anywhere so I'll remove it from there to avoid any confusion.